controllers/components/modelmeshserving/modelmeshserving_actions.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package modelmeshserving
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	componentsv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+	odhdeploy "github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+)
+
+func initialize(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	// early exist
+	_, ok := rr.Instance.(*componentsv1.ModelMeshServing)
+	if !ok {
+		return fmt.Errorf("resource instance %v is not a componentsv1.ModelMeshServing)", rr.Instance)
+	}
+	rr.Manifests = []odhtypes.ManifestInfo{}
+	// setup Manifets[0] for modelmeshserving
+	rr.Manifests = append(rr.Manifests, odhtypes.ManifestInfo{
+		Path:       odhdeploy.DefaultManifestPath,
+		ContextDir: ComponentName,
+		SourcePath: "overlays/odh",
+	})
+	return nil
+}
+
+func devFlags(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	mm, ok := rr.Instance.(*componentsv1.ModelMeshServing)
+	if !ok {
+		return fmt.Errorf("resource instance %v is not a componentsv1.ModelMeshServing)", rr.Instance)
+	}
+
+	if mm.Spec.DevFlags == nil {
+		return nil
+	}
+	// Implement devflags support logic
+	// If dev flags are set, update default manifests path
+	if len(mm.Spec.DevFlags.Manifests) != 0 {
```

Comment:
A common pattern is to try to align the happy path on the left, so I would do something like:

```go
if len(mc.Spec.DevFlags.Manifests) == 0 {
    return nil
}
```